tests/Aspire.Hosting.Tests/HealthCheckTests.cs

```
@@ -29,7 +29,7 @@ public async Task WithHttpHealthCheckThrowsIfReferencingEndpointThatIsNotHttpSch
         var ex = await Assert.ThrowsAsync<DistributedApplicationException>(async () =>
         {
             await app.StartAsync();
-        }).DefaultTimeout();
+        }).DefaultTimeout(TestConstants.DefaultOrchestratorTestTimeout);
```

Are extra constants needed? LongTimeoutDuration is 20 seconds locally and 120 seconds in CI. That seams like it is enough. What takes over 20 seconds locally? We could make the long duration longer if needed.

I don't know what is the right thing for Default, Long (and a new bigger value if required?) timeout consts vs having values explicitly for one product area. I worry that someone will go though and place orchestrator timeouts everywhere just because they're writing code in that area vs using the defaults which suit them.

---

the right timeout when debugging is none 😄

---

https://github.com/dotnet/aspire/blob/0f8b4c1ad94d1a1973269abf5a9bc4db34c73f6e/tests/Shared/AsyncTestHelpers.cs#L98-L103

😄

---

Lets jut increase the default to something high

---

The default timeout being high on the server is fine, and the long timeout being high is fine locally and on the server, but when running tests you don't want to wait 60 seconds because something will never happen. It's a productivity sink.

Default timeout should fail fast in 5 seconds to not waste the developers time.

---

No it should waste developers time. It should never happened locally and when it does there are lots tools you can use.

if you don’t make it high then we won’t be using it in certain classes of tests and it’ll defeat the entire purpose

---

The primary goal of this change is to eliminate false negatives when I am verifying a change by running multiple tests from the IDE, via TestExplorer. With the current timeout values, on my daily go-to dev machine (64 GB RAM, SSD drive, not too old) running `DistributedApplicationTests` results in a handful of tests timing out pretty much **on every run**.  I am wasting a lot of time re-running these failed tests.

Actual test debugging is not an issue with longer timeouts in my experience. When I need a tight inner loop for debugging a particular test, I can just lower/extend individual timeouts as necessary; that scenario happens much less often than "run all test to verify nothing is broken", so I want to optimize for the latter.

There are many reasons why 5 seconds for "short" and 20 seconds for "long" may not be enough for tests that involve DCP. For example, during startup DCP is invoked not 1, but 3 times and does probing for container runtime. If container runtime is "sleeping" that probe may take a while. Anything involving a container is subject to image download/image build delays, and involves retries that DCP does to work-around container orchestrator flakiness. Tests running in parallel affecting each other. Antivirus having to scan new build of Aspire/DCP. Etc. etc.

---

> No it should waste developers time. It should “never” happened locally and when it does there are lots tools you can use (like a debugger)
> 
> if you don’t make it high then we won’t be using it in certain classes of tests and it’ll defeat the entire purpose

Yeah I know that timeouts should never happen when tests are in a good state, but deving code can break things.

When you run tests locally to double check things are good, you want broken tests to fail in 5 seconds with a timeout rather than a minute.